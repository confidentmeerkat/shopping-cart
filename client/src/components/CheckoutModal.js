import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import Modal from 'react-modal';
import RaisedButton from 'material-ui/RaisedButton';
import '../styles/CheckoutModal.css';

export const CheckoutModal = (props) => (
  <Modal
    className="checkout-modal"
    isOpen={props.isOpen}
    onRequestClose={() => props.toggle('checkoutModalOpen')}
  >
    <div className="order">
      <h1>Checkout Information</h1>
      <p>
        <i>Please read the list of items in your order and click "Confirm" to confirm your order.</i>
      </p>
      <table>
        <thead>
          <tr>
            <th>Product Name</th>
            <th>Price</th>
            <th>Quantity</th>
            <th>Total</th>
          </tr>
        </thead>
        <tbody>
          {props.cart.length && props.cart.map((item) => {
            return (
              <tr key={item.product.info.name} >
                <td>{item.product.info.name}</td>
                <td>{numeral(item.product.info.price).format('$0,0.00')}</td>
                <td>{item.quantity}</td>
                <td>{numeral(item.product.info.price * item.quantity).format('$0,0.00')}</td>
              </tr>
            )
          })}
        </tbody>
      </table>
      <p className="total">
        <b>TOTAL AMOUNT: </b>
        <span>{numeral(props.cart.length && props.cart.reduce((acc, item) => acc += item.product.info.price * item.quantity, 0)).format('$0,0.00')}</span>
      </p>
      <div className="btns">
        <RaisedButton
          className="btn"
          label="Cancel"
          primary={true}
          onClick={() => props.toggle('checkoutModalOpen')}
        />
        <RaisedButton
          className="btn"
          label="Confirm"
          backgroundColor="#64DD17"
          labelColor="#fff"
          onClick={props.makeOrder}
        />
      </div>
    </div>
  </Modal>
);

const mapStateToProps = (state) => ({
  cart: state.cart.items
});

export default connect(mapStateToProps)(CheckoutModal);