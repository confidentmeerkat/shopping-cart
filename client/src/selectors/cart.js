export const selectCart = (state) => state.cart;
export const selectItems = (state) => state.cart.items;