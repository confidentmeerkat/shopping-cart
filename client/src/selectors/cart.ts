import { IState, ICart, ICartProduct } from '@typings/state/index';

export const selectCart = (state: IState): ICart => state.cart;
export const selectItems = (state: IState): ICartProduct[] => state.cart.items;
