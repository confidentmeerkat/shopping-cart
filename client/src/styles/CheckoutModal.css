.ReactModalPortal .ReactModal__Overlay--after-open {
  background-color: #00000073 !important;
}

.checkout-modal {
  width: 90%;
  max-width: 500px;
  height: auto;
  margin: auto;
  margin-top: 100px;
  background: white;
  border: 1px solid white;
  border-radius: 5px;
  outline: none;
}

.checkout-modal .order {
  width: 90%;
  max-width: 450px;
  margin: auto;
  margin-top: 10px;
  text-align: center;
  font-family: 'Roboto';
}

.checkout-modal h1 {
  margin-bottom: 60px;
  color: #00BCD4;
}

.checkout-modal table {
  width: 100%;
  margin-top: 40px;
  border-spacing: 0;
}

.checkout-modal th {
  padding: 5px 5px;
  background-color: #00BCD4;
  color: white;
  text-align: left;
}

.checkout-modal td {
  padding: 12px 5px;
  text-align: left;
}

.checkout-modal .total {
  text-align: left;
}

.checkout-modal .total span {
  font-size: 26px;
  font-weight: bold;
  color: #64DD17;
}

.checkout-modal .btns {
  margin-top: 40px;
  margin-bottom: 20px;
}

.checkout-modal .btns .btn {
  margin: 0 20px;
}

@media (max-height: 700px) {
  .checkout-modal {
    height: 500px;
    overflow: scroll;
  }
}