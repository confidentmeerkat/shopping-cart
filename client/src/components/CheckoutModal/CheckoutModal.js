import React from 'react';
import numeral from 'numeral';
import Modal from 'react-modal';
import RaisedButton from 'material-ui/RaisedButton';
import '@styles/CheckoutModal.css';

export const CheckoutModal = ({ cart, isOpen, toggle, makeOrder }) => (
  <Modal
    className="checkout-modal"
    isOpen={isOpen}
    onRequestClose={() => toggle('checkoutModalOpen')}
  >
    <div className="order">
      <h1>Checkout Information</h1>
      <p>
        <i>Please read the list of items in your order and click "Confirm" to confirm your order.</i>
      </p>
      <table>
        <thead>
          <tr>
            <th>Product Name</th>
            <th>Price</th>
            <th>Quantity</th>
            <th>Total</th>
          </tr>
        </thead>
        <tbody>
          {cart.length && cart.map((item) => {
            return (
              <tr key={item.product.info.name} >
                <td>{item.product.info.name}</td>
                <td>{numeral(item.product.info.price).format('$0,0.00')}</td>
                <td>{item.quantity}</td>
                <td>{numeral(item.product.info.price * item.quantity).format('$0,0.00')}</td>
              </tr>
            )
          })}
        </tbody>
      </table>
      <p className="total">
        <b>TOTAL AMOUNT: </b>
        <span>{numeral(cart.length && cart.reduce((acc, item) => acc += item.product.info.price * item.quantity, 0)).format('$0,0.00')}</span>
      </p>
      <div className="btns">
        <RaisedButton
          className="btn"
          label="Cancel"
          primary={true}
          onClick={() => toggle('checkoutModalOpen')}
        />
        <RaisedButton
          className="btn"
          label="Confirm"
          backgroundColor="#64DD17"
          labelColor="#fff"
          onClick={makeOrder}
        />
      </div>
    </div>
  </Modal>
);

export default CheckoutModal;
