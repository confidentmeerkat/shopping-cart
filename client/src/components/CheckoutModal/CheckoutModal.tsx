import * as React from 'react';
import * as numeral from 'numeral';
import * as Modal from 'react-modal';
import { ICartProduct } from '@typings/state/index';
import { ModalProps } from '@typings/modal';
import Button from '@material-ui/core/Button';
import '@styles/CheckoutModal.css';

const CheckoutModal: React.SFC<ModalProps> = ({ cart, isOpen, setActiveModal, makeOrder }): JSX.Element => (
  <Modal
    className="checkout-modal"
    isOpen={isOpen}
    onRequestClose={() => setActiveModal(null)}
  >
    <div className="order">
      <h1>Checkout Information</h1>
      <p>
        <i>Please read the list of items in your order and click "Confirm" to confirm your order.</i>
      </p>
      <table>
        <thead>
          <tr>
            <th>Product Name</th>
            <th>Price</th>
            <th>Quantity</th>
            <th>Total</th>
          </tr>
        </thead>
        <tbody>
          {cart!.length && cart!.map((item: ICartProduct) => {
            return (
              <tr key={item.product.info.name} >
                <td>{item.product.info.name}</td>
                <td>{numeral(item.product.info.price).format('$0,0.00')}</td>
                <td>{item.quantity}</td>
                <td>{numeral(item.product.info.price * item.quantity!).format('$0,0.00')}</td>
              </tr>
            )
          })}
        </tbody>
      </table>
      <p className="total">
        <b>TOTAL AMOUNT: </b>
        <span>{numeral(cart!.length && cart!.reduce((acc, item) => acc += item.product.info.price * item.quantity!, 0)).format('$0,0.00')}</span>
      </p>
      <div className="btns">
        <Button
          className="btn"
          onClick={() => setActiveModal(null)}>
          Cancel
        </Button>
        <Button
          className="btn"
          onClick={makeOrder}>
          Confirm
        </Button>
      </div>
    </div>
  </Modal>
);

export default CheckoutModal;
