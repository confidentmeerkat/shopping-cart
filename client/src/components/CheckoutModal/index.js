import { compose } from 'recompose';
import { connect } from 'react-redux';
import CheckoutModal from './CheckoutModal';

const mapStateToProps = (state) => ({
  cart: state.cart.items
});

export default compose(
  connect(mapStateToProps)
)(CheckoutModal);
