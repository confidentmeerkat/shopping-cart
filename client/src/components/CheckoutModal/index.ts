import { compose } from 'recompose';
import { connect } from 'react-redux';
import { selectItems } from '@selectors/cart';
import { IState, IProduct, } from '@state/index';
import { Props } from './CheckoutModal';
import CheckoutModal from './CheckoutModal';

interface EnhancedProps extends Props {
  cart?: IProduct[];
}

const mapStateToProps = (state: IState) => ({
  cart: selectItems(state)
});

export default compose<Props, EnhancedProps>(
  connect(mapStateToProps)
)(CheckoutModal);
