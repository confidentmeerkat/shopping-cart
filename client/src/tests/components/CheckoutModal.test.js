import React from 'react';
import { shallow } from 'enzyme';
import { CheckoutModal } from '../../components/CheckoutModal';
import cart from '../__mocks__/cart';

describe('<CheckoutModal />', () => {
  it('renders CheckoutModal correctly', () => {
    const wrapper = shallow(<CheckoutModal
      isOpen={true}
      onRequestClose={() => { }}
      toggle={() => { }}
      makeOrder={() => { }}
      cart={cart.items}
    />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('tbody').children()).toHaveLength(2);
    expect(wrapper.find('span').text()).toBe('$2,200.00');
  });
});