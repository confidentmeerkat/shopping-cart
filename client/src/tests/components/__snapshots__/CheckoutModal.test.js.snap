// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckoutModal /> renders CheckoutModal correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="checkout-modal"
  closeTimeoutMS={0}
  isOpen={true}
  onRequestClose={[Function]}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <div
    className="order"
  >
    <h1>
      Checkout Information
    </h1>
    <p>
      <i>
        Please read the list of items in your order and click "Confirm" to confirm your order.
      </i>
    </p>
    <table>
      <thead>
        <tr>
          <th>
            Product Name
          </th>
          <th>
            Price
          </th>
          <th>
            Quantity
          </th>
          <th>
            Total
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          key="Apple iPhone"
        >
          <td>
            Apple iPhone
          </td>
          <td>
            $700.00
          </td>
          <td>
            1
          </td>
          <td>
            $700.00
          </td>
        </tr>
        <tr
          key="Samsung Galaxy S8"
        >
          <td>
            Samsung Galaxy S8
          </td>
          <td>
            $750.00
          </td>
          <td>
            2
          </td>
          <td>
            $1,500.00
          </td>
        </tr>
      </tbody>
    </table>
    <p
      className="total"
    >
      <b>
        TOTAL AMOUNT: 
      </b>
      <span>
        $2,200.00
      </span>
    </p>
    <div
      className="btns"
    >
      <RaisedButton
        className="btn"
        disabled={false}
        fullWidth={false}
        label="Cancel"
        labelPosition="after"
        onClick={[Function]}
        primary={true}
        secondary={false}
      />
      <RaisedButton
        backgroundColor="#64DD17"
        className="btn"
        disabled={false}
        fullWidth={false}
        label="Confirm"
        labelColor="#fff"
        labelPosition="after"
        onClick={[Function]}
        primary={false}
        secondary={false}
      />
    </div>
  </div>
</Modal>
`;
